### PR TITLE
migrate stuff from physics benchmarks to detector benchmarks

### PR DESCRIPTION
**Pull request by sebouh137**
### Briefly, what does this PR introduce?
Adds benchmarks that were in physics benchmarks to detector benchmarks:
- lambda and sigma in zdc
- neutron in insert

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [X] Other: __

### Please check if this PR fulfills the following:
- [x] Tests for the changes have been added
- [x] Documentation has been added / updated
- [X] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
no
### Does this PR change default behavior?
yes.  Runs benchmarks that were previously in the other repository

### CODE REVIEW

**Comment by sebouh137** [comment]
When migrating these benchmarks to the detector benchmarks, I noticed that there is already a "zdc" benchmark;  I'm not sure if this was based on an older implementation of the ZDC, and if so, it should be deprecated in favor of newer benchmarks.

File: benchmarks/zdc_lambda/config.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      - P: 275
+  timeout: 3 hours
+  script:
+    - snakemake --cores 1 sim_output/zdc_lambda/epic_zdc_sipm_on_tile_only_rec_lambda_dec_${P}GeV.edm4hep.root 
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+
+analyze:zdc_lambda
```
Comment (veprbl) [inline benchmarks/zdc_lambda/config.yml:23]: The error reporting in github trigger action is not working, but you can be sure that issue is gitlab syntax
https://eicweb.phy.anl.gov/EIC/benchmarks/detector_benchmarks/-/pipelines/102784

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -44,6 +44,7 @@ stages:
   - simulate
   - calibrate
   - benchmarks
+  - analyze
```
Comment (veprbl) [inline .gitlab-ci.yml:47]: What is `analyze` in physics_benchmarks is `benchmarks` here.
```suggestion
```

File: benchmarks/zdc_lambda/config.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+simulate:zdc_lambda:
```
Comment (veprbl) [inline benchmarks/zdc_lambda/config.yml:1]:
```suggestion
sim:zdc_lambda:
```

File: benchmarks/zdc_sigma/config.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+simulate:zdc_sigma:
```
Comment (veprbl) [inline benchmarks/zdc_sigma/config.yml:1]:
```suggestion
sim:zdc_sigma:
```

File: benchmarks/zdc_sigma/config.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    - snakemake --cores 1 sim_output/zdc_sigma/epic_zdc_sipm_on_tile_only_rec_sigma_dec_${P}GeV.edm4hep.root 
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+
+analyze:zdc_sigma:
+  stage: analyze
+  extends: .phy_benchmark
+  needs: ["simulate:zdc_sigma"]
```
Comment (veprbl) [inline benchmarks/zdc_sigma/config.yml:26]:
```suggestion
  needs: ["sim:zdc_sigma"]
```

File: benchmarks/insert_neutron/config.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
+simulate:insert_neutron:
```
Comment (veprbl) [inline benchmarks/insert_neutron/config.yml:1]:
```suggestion
sim:insert_neutron:
```

File: benchmarks/insert_neutron/config.yml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    - snakemake --cores 1 sim_output/insert_neutron/epic_craterlake_rec_neutron_${P}GeV.edm4hep.root 
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+
+analyze:insert_neutron:
+  stage: analyze
+  extends: .phy_benchmark
+  needs: ["simulate:insert_neutron"]
```
Comment (veprbl) [inline benchmarks/insert_neutron/config.yml:25]:
```suggestion
  needs: ["sim:insert_neutron"]
```

File: benchmarks/zdc_lambda/config.yml [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    - snakemake --cores 1 sim_output/zdc_lambda/epic_zdc_sipm_on_tile_only_rec_lambda_dec_${P}GeV.edm4hep.root 
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+
+analyze:zdc_lambda
+  stage: analyze
+  extends: .phy_benchmark
+  needs: ["simulate:zdc_lambda"]
```
Comment (veprbl) [inline benchmarks/zdc_lambda/config.yml:26]:
```suggestion
  needs: ["sim:zdc_lambda"]
```

**Comment by veprbl** [comment]
> When migrating these benchmarks to the detector benchmarks, I noticed that there is already a "zdc" benchmark; I'm not sure if this was based on an older implementation of the ZDC, and if so, it should be deprecated in favor of newer benchmarks.

Feel free to remove it, if you feel like yours is better.

File: benchmarks/zdc_lambda/config.yml (hunk 1 above)
Comment (veprbl) [reply benchmarks/zdc_lambda/config.yml:23]:
```suggestion
analyze:zdc_lambda:
```

File: benchmarks/insert_neutron/config.yml [hunk 9]
```diff
@@ -0,0 +1,34 @@
+sim:insert_neutron:
+  stage: simulate
+  extends: .phy_benchmark
```
Comment (veprbl) [inline benchmarks/insert_neutron/config.yml:3]:
```suggestion
  extends: .det_benchmark
```

File: benchmarks/insert_neutron/config.yml [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+  script:
+    - snakemake --cores 1 sim_output/insert_neutron/epic_craterlake_rec_neutron_${P}GeV.edm4hep.root 
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+
+analyze:insert_neutron:
+  stage: analyze
+  extends: .phy_benchmark
```
Comment (veprbl) [inline benchmarks/insert_neutron/config.yml:24]:
```suggestion
  extends: .det_benchmark
```

File: benchmarks/insert_neutron/config.yml [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+  needs: ["sim:insert_neutron"]
+  script:
+    - mkdir -p results/epic_craterlake
+    - python benchmarks/insert_neutron/analysis/neutron_plots.py results/epic_craterlake/insert_neutron
+
+collect_results:insert_neutron:
+  stage: collect_results
```
Comment (veprbl) [inline benchmarks/insert_neutron/config.yml:31]:
```suggestion
  stage: collect
```

File: benchmarks/zdc_lambda/config.yml [hunk 12]
```diff
@@ -0,0 +1,34 @@
+sim:zdc_lambda:
+  stage: simulate
+  extends: .phy_benchmark
```
Comment (veprbl) [inline benchmarks/zdc_lambda/config.yml:3]:
```suggestion
  extends: .det_benchmark
```

File: benchmarks/zdc_lambda/config.yml [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+  script:
+    - snakemake --cores 1 sim_output/zdc_lambda/epic_zdc_sipm_on_tile_only_rec_lambda_dec_${P}GeV.edm4hep.root 
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+
+analyze:zdc_lambda:
+  stage: analyze
+  extends: .phy_benchmark
```
Comment (veprbl) [inline benchmarks/zdc_lambda/config.yml:25]:
```suggestion
  extends: .det_benchmark
```

File: benchmarks/zdc_lambda/config.yml [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+  needs: ["sim:zdc_lambda"]
+  script:
+    - snakemake --cores 1 results/epic_zdc_sipm_on_tile_only/zdc_lambda 
+
+collect_results:zdc_lambda:
+  stage: collect_results
```
Comment (veprbl) [inline benchmarks/zdc_lambda/config.yml:31]:
```suggestion
  stage: collect
```

File: benchmarks/zdc_sigma/config.yml [hunk 15]
```diff
@@ -0,0 +1,34 @@
+sim:zdc_sigma:
+  stage: simulate
+  extends: .phy_benchmark
```
Comment (veprbl) [inline benchmarks/zdc_sigma/config.yml:3]:
```suggestion
  extends: .det_benchmark
```

File: benchmarks/zdc_sigma/config.yml [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+  script:
+    - snakemake --cores 1 sim_output/zdc_sigma/epic_zdc_sipm_on_tile_only_rec_sigma_dec_${P}GeV.edm4hep.root 
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+
+analyze:zdc_sigma:
+  stage: analyze
+  extends: .phy_benchmark
```
Comment (veprbl) [inline benchmarks/zdc_sigma/config.yml:25]:
```suggestion
  stage: benchmark
  extends: .det_benchmark
```

File: benchmarks/zdc_sigma/config.yml [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+  needs: ["sim:zdc_sigma"]
+  script:
+    - snakemake --cores 1 results/epic_zdc_sipm_on_tile_only/zdc_sigma 
+
+collect_results:zdc_sigma:
+  stage: collect_results
```
Comment (veprbl) [inline benchmarks/zdc_sigma/config.yml:31]:
```suggestion
  stage: collect
```

File: benchmarks/insert_neutron/config.yml [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+  timeout: 1 hours
+  script:
+    - snakemake --cores 1 sim_output/insert_neutron/epic_craterlake_rec_neutron_${P}GeV.edm4hep.root 
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+
+analyze:insert_neutron:
+  stage: analyze
```
Comment (veprbl) [inline benchmarks/insert_neutron/config.yml:23]:
```suggestion
  stage: benchmark
```

File: benchmarks/zdc_lambda/config.yml [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+  timeout: 3 hours
+  script:
+    - snakemake --cores 1 sim_output/zdc_lambda/epic_zdc_sipm_on_tile_only_rec_lambda_dec_${P}GeV.edm4hep.root 
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+
+analyze:zdc_lambda:
+  stage: analyze
```
Comment (veprbl) [inline benchmarks/zdc_lambda/config.yml:24]:
```suggestion
  stage: benchmark
```

File: benchmarks/insert_neutron/config.yml [hunk 20]
```diff
@@ -0,0 +1,34 @@
...
+  timeout: 1 hours
+  script:
+    - snakemake --cores 1 sim_output/insert_neutron/epic_craterlake_rec_neutron_${P}GeV.edm4hep.root 
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+
+analyze:insert_neutron:
+  stage: benchmark
```
Comment (veprbl) [inline benchmarks/insert_neutron/config.yml:23]:
```suggestion
  stage: benchmarks
```

File: benchmarks/zdc_lambda/config.yml [hunk 21]
```diff
@@ -0,0 +1,34 @@
...
+  timeout: 3 hours
+  script:
+    - snakemake --cores 1 sim_output/zdc_lambda/epic_zdc_sipm_on_tile_only_rec_lambda_dec_${P}GeV.edm4hep.root 
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+
+analyze:zdc_lambda:
+  stage: benchmark
```
Comment (veprbl) [inline benchmarks/zdc_lambda/config.yml:24]:
```suggestion
  stage: benchmarks
```

File: benchmarks/zdc_sigma/config.yml [hunk 22]
```diff
@@ -0,0 +1,34 @@
...
+  timeout: 6 hours
+  script:
+    - snakemake --cores 1 sim_output/zdc_sigma/epic_zdc_sipm_on_tile_only_rec_sigma_dec_${P}GeV.edm4hep.root 
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+
+analyze:zdc_sigma:
+  stage: benchmark
```
Comment (veprbl) [inline benchmarks/zdc_sigma/config.yml:24]:
```suggestion
  stage: benchmarks
```

File: benchmarks/insert_neutron/config.yml [hunk 23]
```diff
@@ -0,0 +1,34 @@
...
+      - P: 80
+  timeout: 1 hours
+  script:
+    - snakemake --cores 1 sim_output/insert_neutron/epic_craterlake_rec_neutron_${P}GeV.edm4hep.root 
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+
+analyze:insert_neutron:
```
Comment (veprbl) [inline benchmarks/insert_neutron/config.yml:22]:
```suggestion
bench:insert_neutron:
```

File: benchmarks/zdc_lambda/config.yml [hunk 24]
```diff
@@ -0,0 +1,34 @@
...
+      - P: 275
+  timeout: 3 hours
+  script:
+    - snakemake --cores 1 sim_output/zdc_lambda/epic_zdc_sipm_on_tile_only_rec_lambda_dec_${P}GeV.edm4hep.root 
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+
+analyze:zdc_lambda:
```
Comment (veprbl) [inline benchmarks/zdc_lambda/config.yml:23]:
```suggestion
bench:zdc_lambda:
```

File: benchmarks/zdc_sigma/config.yml [hunk 25]
```diff
@@ -0,0 +1,34 @@
...
+      - P: 275
+  timeout: 6 hours
+  script:
+    - snakemake --cores 1 sim_output/zdc_sigma/epic_zdc_sipm_on_tile_only_rec_sigma_dec_${P}GeV.edm4hep.root 
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+
+analyze:zdc_sigma:
```
Comment (veprbl) [inline benchmarks/zdc_sigma/config.yml:23]:
```suggestion
bench:zdc_sigma:
```

File: benchmarks/insert_neutron/config.yml [hunk 26]
```diff
@@ -0,0 +1,34 @@
...
+  stage: benchmarks
+  extends: .det_benchmark
+  needs: ["sim:insert_neutron"]
+  script:
+    - mkdir -p results/epic_craterlake
+    - python benchmarks/insert_neutron/analysis/neutron_plots.py results/epic_craterlake/insert_neutron
+
+collect_results:insert_neutron:
+  stage: collect
+  needs: ["analyze:insert_neutron"]
```
Comment (veprbl) [inline benchmarks/insert_neutron/config.yml:32]:
```suggestion
  needs: ["bench:insert_neutron"]
```

File: benchmarks/zdc_lambda/config.yml [hunk 27]
```diff
@@ -0,0 +1,34 @@
...
+bench:zdc_lambda:
+  stage: benchmarks
+  extends: .det_benchmark
+  needs: ["sim:zdc_lambda"]
+  script:
+    - snakemake --cores 1 results/epic_zdc_sipm_on_tile_only/zdc_lambda 
+
+collect_results:zdc_lambda:
+  stage: collect
+  needs: ["analyze:zdc_lambda"]
```
Comment (veprbl) [inline benchmarks/zdc_lambda/config.yml:32]:
```suggestion
  needs: ["bench:zdc_lambda"]
```

File: benchmarks/zdc_sigma/config.yml [hunk 28]
```diff
@@ -0,0 +1,34 @@
...
+bench:zdc_sigma:
+  stage: benchmarks
+  extends: .det_benchmark
+  needs: ["sim:zdc_sigma"]
+  script:
+    - snakemake --cores 1 results/epic_zdc_sipm_on_tile_only/zdc_sigma 
+
+collect_results:zdc_sigma:
+  stage: collect
+  needs: ["analyze:zdc_sigma"]
```
Comment (veprbl) [inline benchmarks/zdc_sigma/config.yml:32]:
```suggestion
  needs: ["bench:zdc_sigma"]
```

File: benchmarks/insert_neutron/config.yml
Comment (sebouh137) [inline benchmarks/insert_neutron/config.yml:1]: this change, along with the analogous ones in the other benchmarks, causes them to crash.

Comment (veprbl) [reply benchmarks/insert_neutron/config.yml:1]: Right, but it's not supposed to. I will figure it out.

**Review by veprbl — APPROVED** [review]

File: benchmarks/insert_neutron/config.yml
Comment (veprbl) [reply benchmarks/insert_neutron/config.yml:1]: It looks like after we've arrived at faster benchmarks the issue has gone away.

The outputs of ls -la somehow always show up some very recent modification times. I'm wondering why those are not coming from git. It also looks like gitlab's procedure is to make a copy of a work directory and then do `git clean` on it. There is an option to do fresh clones instead, that's an option. Or, maybe, we need to apply https://github.com/MestreLion/git-tools/blob/main/git-restore-mtime to correct the timestamps.

**Pull request merged**